src/socket.c

```diff
@@ -969,6 +969,15 @@ void pthreads_socket_get_last_error(zval *object, zend_bool clear, zval *return_
 	}
 }
 
+void pthreads_socket_strerror(zend_long error, zval *return_value) {
+	char *errstr;
+	errstr = php_socket_strerror(error, NULL, 0);
+
+	RETVAL_STRING(errstr);
+
+	efree(errstr);
+}
+
```

Comment:
@tpunt do you see any problem here with the alloc(`php_socket_strerror`) / efree() or is this part fine? `php_socket_strerror` returns an `char*` in every case, so it shouldn't be necessary to check for null, I think.

Reply:
This seems fine to me. `errstr` will always be ZMM-allocated if `NULL` is passed in as the buffer, and `RETVAL_STRING` will internally perform a copy the returned `char *` buffer, so `errstr` does indeed need to be freed afterwards.